python/lsst/daf/butler/core/datasets/ref.py

```diff
@@ -152,6 +158,103 @@ def __lt__(self, other: Any) -> bool:
         # Compare tuples in the priority order
         return (self_run, self.datasetType, self.dataId) < (other_run, other.datasetType, other.dataId)
 
+    def to_simple(self, minimal: bool = False) -> Dict:
+        """Convert this class to a simple python type suitable for
+        serialization.
+
+        Parameters
+        ----------
+        minimal : `bool`, optional
+            Use minimal serialization. Requires Registry to convert
+            back to a full type.
+
+        Returns
+        -------
+        simple : `dict` or `int`
+            The object converted to a dictionary.
+        """
+        if minimal and self.id is not None:
+            # The only thing needed to uniquely define a DatasetRef
+            # is the integer id so that can be used directly if it is
+            # resolved and if it is not a component DatasetRef.
+            # Store is in a dict to allow us to easily add the planned
+            # origin information later without having to support
+            # an int and dict in simple form.
+            simple: Dict[str, Any] = {"id": self.id}
+            if self.isComponent():
+                # We can still be a little minimalist with a component
+                # but we will also need to record the datasetType component
+                simple["component"] = self.datasetType.component()
+            return simple
+
+        # Convert to a dict form
+        as_dict: Dict[str, Any] = {"datasetType": self.datasetType.to_simple(minimal=minimal),
+                                   "dataId": self.dataId.to_simple(),
+                                   "hasParentId": self.hasParentId
```

Comment:
@TallJimbo I'm fairly certain that this is a left over for when we first changed the handling of components in registry and it was possible to have a component in registry or in datastore depending on how the put had been implemented.

It was added in #269 and I don't see any code that checks the value anywhere.

Reply:
I don't recall anything that uses it (I've forgotten the details how we identify component dataset refs), so I'm all for removing if it doesn't break anything.

Reply:
I removed it and everything is still working...